include/utils/system_reference_blas.hpp

```diff
@@ -28,13 +28,27 @@
 
 #include "cblas.h"
 #include <iostream>
+#include <cmath>
 
-extern "C" {
-// i*amin is an extension
-CBLAS_INDEX cblas_isamin(const int N, const float *X, const int incX);
-CBLAS_INDEX cblas_idamin(const int N, const double *X, const int incX);
-CBLAS_INDEX cblas_icamin(const int N, const void *X, const int incX);
-CBLAS_INDEX cblas_izamin(const int N, const void *X, const int incX);
+// i*amin is an extension, provide an implementation
+template<typename scalar_t>
```

Comment:
Can this be in an anonymous namespace if it's not public?

Reply:
Done